Automattic/wp-calypso
# Extract `DiscoverNavigation` to `ScrollableHorizontalNavigation`

Related to https://github.com/Automattic/dotcom-forge/issues/7537

## Proposed Changes

* Extract the navigation logic from `DiscoverNavigation` into the new `ScrollableHorizontalNavigation` component, so it can be reused.

## Why are these changes being made?

* On https://github.com/Automattic/dotcom-forge/issues/7537 it was suggested that we could reuse the navigation on https://wordpress.com/discover for the changes needed on https://wordpress.com/plugins. So, instead of copy-pasting that code, I'm extracting it.

## Testing Instructions

* Open the live preview and go to `/discover`
* Check that the navigation still works the same as on https://wordpress.com/discover

## Pre-merge Checklist
